cloudigrade/account/tests/test_tasks.py

```diff
@@ -0,0 +1,128 @@
+"""Collection of tests for celery tasks."""
+import random
+import uuid
+from unittest.mock import patch
+
+from celery.exceptions import Retry
+from django.test import TestCase
+
+from account.models import AwsAccount, AwsMachineImage
+from account.tasks import copy_ami_snapshot
+from util.exceptions import (AwsSnapshotCopyLimitError,
+                             AwsSnapshotEncryptedError,
+                             AwsSnapshotNotOwnedError)
+from util.tests import helper as util_helper
+
+
+class AccountCeleryTaskTest(TestCase):
+    """Account app Celery task test cases."""
+
+    @patch('account.tasks.aws')
+    def test_copy_ami_snapshot_success(self, mock_aws):
+        """Assert that the snapshot copy task succeeds."""
+        mock_arn = util_helper.generate_dummy_arn()
+        mock_region = random.choice(util_helper.SOME_AWS_REGIONS)
+        mock_image_id = str(uuid.uuid4())
+        mock_image = util_helper.generate_mock_image(mock_image_id)
+        block_mapping = mock_image.block_device_mappings
+        mock_snapshot_id = block_mapping[0]['Ebs']['SnapshotId']
+        mock_new_snapshot_id = util_helper.generate_mock_snapshot_id()
+        mock_session = mock_aws.boto3.Session.return_value
+
+        mock_aws.get_session.return_value = mock_session
+        mock_aws.get_ami.return_value = mock_image
+        mock_aws.get_ami_snapshot_id.return_value = mock_snapshot_id
+        mock_aws.copy_snapshot.return_value = mock_new_snapshot_id
+
+        copy_ami_snapshot(mock_arn, mock_image_id, mock_region)
+
+        mock_aws.get_session.asssert_called_with(mock_arn)
+        mock_aws.get_ami.asssert_called_with(
+            mock_session,
+            mock_image_id,
+            mock_region
+        )
+        mock_aws.get_ami_snapshot_id.asssert_called_with(mock_image)
```

Comment:
Should that be "assert_called_with" ?

Reply:
Yes. Thanks mocking for letting that fly... ☠️